### PR TITLE
DT-20651 Improve find forms search string comparison for better fault tolerance.

### PR DESCRIPTION
**Pull request by zacharymorel**
## Description
In its current form, the Find a VA Forms API does simple string comparison between the search term entered and the form name and title. It splits up the terms entered which can yield to poor results when the user does not include dashes, or uses a partial form name.

**Important:** We're temporarily solving this issue on the FE since the dependency on other teams for the permanent solve is more complex and time consuming. 

## Testing done
Updated tests, ran e2e, unit, integration tests. Manually testing running locally.

## Screenshots
Screen Shot 2021-03-05 at 1 25 25 PM

## Acceptance criteria
- [x] The query has fault tolerance.
- [x] GA event recorder is moved so we still record user entered value.
- [x] Solution has room for additional values to be added in the matrix.

## Definition of done
- [x] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/find-forms/actions/index.js [hunk 1]
```diff
@@ -119,6 +116,14 @@ export const fetchFormsThunk = (query, options = {}) => async dispatch => {
...
       'type-ahead-options-list': undefined, // populate with undefined since type ahead won't feature here
     });
+
+    // If we are here, the API request succeeded.
+    dispatch(
+      fetchFormsSuccess(
+        resultsDetails.results,
+        resultsDetails.hasOnlyRetiredForms,
+      ),
+    );
```
Comment (zacharymorel) [inline src/applications/find-forms/actions/index.js:126]: I just moved this down.

File: src/applications/find-forms/tests/helpers/index.unit.spec.js [hunk 2]
```diff
@@ -113,4 +117,14 @@ describe('Find VA Forms helpers', () => {
...
+
+  it('regexpDashAdder adds the dash where it should', () => {
+    const transformString = regexpDashAdder('20213', 2);
+    expect(transformString).to.eql('20-213');
+  });
+
+  it('transformSearchTerm adjusts the term to allow for faulty searches', () => {
+    const transformString = transformSearchTerm('1010');
+    expect(transformString).to.eql('10-10');
+  });
```
Comment (ncksllvn) [inline src/applications/find-forms/tests/helpers/index.unit.spec.js:129]: The way this function works with a direct input and a direct output makes it golden for a strong unit test. You could list a whole bunch of inputs/output in an array and iterate across them all.

```js
const searchQueryAutoCorrectResults = new Map([
    ['1010', '10-10'],
    // ... tons of inputs/outputs
]);

for (const [term, correctedTerm] of searchQueryAutoCorrectResults) {
    it('produces the expected auto-corrected result', () => {
        const result = transformSearchTerm(term);
        expect(result).to.equal(correctedTerm);
    )
}
```

Great opportunity to have your unit test reflect exactly all of the different input/output edge cases you may have tested with manually

**Review by ncksllvn — COMMENTED** [review]
Left some comments for potential improvements (i think just naming/testing) but this looks REALLY good!

File: src/applications/find-forms/tests/helpers/index.unit.spec.js (hunk 2 above)
Comment (zacharymorel) [reply src/applications/find-forms/tests/helpers/index.unit.spec.js:129]: Oh wow, that's a great point, we can take it a step further!

Comment (zacharymorel) [reply src/applications/find-forms/tests/helpers/index.unit.spec.js:148]: @ncksllvn This is ready for another review when you get the chance.

Comment (ncksllvn) [reply src/applications/find-forms/tests/helpers/index.unit.spec.js:148]: Well done!!

**Review by ncksllvn — APPROVED** [review]
LGTM. Great job!!

**Pull request merged**